test/unit/org/apache/cassandra/io/sstable/MutlipleSSTableFormatsTest.java

```diff
@@ -0,0 +1,173 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.cql3.UntypedResultSet;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.io.sstable.format.SSTableFormat;
+import org.apache.cassandra.io.sstable.format.big.BigFormat;
+import org.apache.cassandra.io.sstable.format.trieindex.TrieIndexFormat;
+import org.assertj.core.api.Assertions;
+
+
+public class MutlipleSSTableFormatsTest extends CQLTester
+{
+    private final static Logger logger = LoggerFactory.getLogger(MutlipleSSTableFormatsTest.class);
+    private final static int cnt = 100;
+    private final static int overlap = 70;
+    private final static int deletionCount = 30;
+
+    private final long seed = System.nanoTime();
+    private Random random;
+
+    private String savedProp;
+
+    @Before
+    public void before() {
+        savedProp = System.getProperty(SSTableFormat.FORMAT_DEFAULT_PROP);
+        random = new Random(seed);
+        logger.info("Using random seed = {}", seed);
+    }
+
+    @After
+    public void after() {
+        if (savedProp == null)
+            System.getProperties().remove(SSTableFormat.FORMAT_DEFAULT_PROP);
+        else
+            System.setProperty(SSTableFormat.FORMAT_DEFAULT_PROP, savedProp);
+    }
+
+    private Map<Integer, Integer> createSSTables() throws Throwable {
+        Map<Integer, Integer> content = Maps.newHashMap();
+
+        createTable("CREATE TABLE %s (id INT, val INT, PRIMARY KEY (id))");
+        disableCompaction();
+
+        int offset = 0;
+        for (SSTableFormat.Type formatType : SSTableFormat.Type.values())
+        {
+            System.setProperty(SSTableFormat.FORMAT_DEFAULT_PROP, formatType.name);
+
+            for (int i = 0; i < cnt; i++)
+            {
+                int v = random.nextInt();
+                content.put(i + offset, v);
+                execute("INSERT INTO %s (id, val) VALUES (?, ?)", i + offset, v);
+            }
+            offset += cnt - overlap;
+
+            flush();
+        }
+
+        for (SSTableFormat.Type formatType : SSTableFormat.Type.values())
+        {
+            System.setProperty(SSTableFormat.FORMAT_DEFAULT_PROP, formatType.name);
+
+            for (int i = 0; i < deletionCount; i++)
+            {
+                int key = random.nextInt(offset + overlap);
+                content.remove(key);
+                execute("DELETE FROM %s WHERE id = ?", key);
+            }
+
+            flush();
+        }
+
+        List<SSTableFormat.Type> createdFormats = createdFormats();
+        Assertions.assertThat(createdFormats).hasSameElementsAs(Sets.newHashSet(SSTableFormat.Type.values()));
+
+        return content;
+    }
+
+    private void checkRead(Map<Integer, Integer> content) throws Throwable {
+        for (Map.Entry<Integer, Integer> entry : content.entrySet())
```

Comment:
this check is redundant. The check below covers this.

Reply:
no, it is not - while the first one uses point queries, the other one uses a range query